web/src/korrel8r/node-factory.ts

```diff
@@ -39,9 +39,9 @@ export class Korrel8rNodeFactory {
     return InvalidNode.fromURL(url);
   }
 
-  static fromQuery(query: string): Korrel8rNode {
+  static fromQuery(query: string, start: number | null, end: number | null): Korrel8rNode {
```

Comment:
```suggestion
  static fromQuery(query: string, constraint?: Constraint): Korrel8rNode {
```
Please change start, end to a single Constraint parameter. That makes it easier to use fromQuery without a constraint and more importantly means we won't have to change the signature if we want to add support for more constraint fields in future (e.g. limit)